### PR TITLE
feat;SomeInputValidation

### PR DESCRIPTION
**Pull request by asbel**

### CODE REVIEW

File: assets/scripts/transactions.js [hunk 1]
```diff
@@ -1,13 +1,39 @@
 const Transaction = require('./model/transaction');
 const jsonUtil = require('./utils/json-util');
 const dateUtil = require('./utils/date-util');
-
+const moment = require('moment');
```
Comment (nickdemianchuk) [inline assets/scripts/transactions.js]: move to a separate file

File: assets/scripts/transactions.js [hunk 2]
```diff
@@ -1,13 +1,39 @@
...
+        this.oldSelectionEnd = this.selectionEnd;
+      } else if (this.hasOwnProperty("oldValue")) {
+        this.value = this.oldValue;
+        this.setSelectionRange(this.oldSelectionStart, this.oldSelectionEnd);
+        event.preventDefault();
+      }
+    });
+  });
+}
+//function numbersOnly()
```
Comment (nickdemianchuk) [inline assets/scripts/transactions.js]: remove the comments

File: assets/scripts/transactions.js [hunk 3]
```diff
@@ -16,7 +42,11 @@ function displayTransaction(transaction) {
     let row = transactionTableBody.insertRow();
 
     let dateColumn = row.insertCell(0);
-    dateColumn.innerHTML = dateUtil.getShortDate(transaction.date);
+    dateColumn = moment(dateColumn);
```
Comment (nickdemianchuk) [inline assets/scripts/transactions.js]: create a new method in 'date-util.js'

File: assets/scripts/transactions.js [hunk 4]
```diff
@@ -25,6 +55,9 @@ function displayTransaction(transaction) {
     categoryColumn.innerHTML = transaction.category;
 
     let amountColumn = row.insertCell(3);
+    let amount = setInputFilter(document.getElementById("amount"), function(value) {
```
Comment (nickdemianchuk) [inline assets/scripts/transactions.js]: create a new util file for that

File: package.json [hunk 5]
```diff
@@ -17,5 +17,7 @@
     "webpack": "^4.29.6",
     "webpack-cli": "^3.3.0"
   },
-  "dependencies": {}
+  "dependencies": {
```
Comment (nickdemianchuk) [inline package.json:20]: move to devDependencies

File: assets/html/bills.html [hunk 6]
```diff
@@ -50,6 +50,7 @@ <h1>Add your own bill</h1>
         <br>
         <input type="button" id="add-button" value="Add">
     </form>
+<script src="./../../dist/transactions.js"></script>
```
Comment (nickdemianchuk) [inline assets/html/bills.html]: should be a separate file for input validation

File: assets/scripts/index.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+const express = require('express')
```
Comment (nickdemianchuk) [inline assets/scripts/index.js]: delete this file

File: settings.gradle [hunk 8]
```diff
@@ -0,0 +1 @@
+include ':app'
```
Comment (nickdemianchuk) [inline settings.gradle:1]: delete this file

File: assets/scripts/transactions.js [hunk 9]
```diff
@@ -1,13 +1,39 @@
...
+var today = moment().calendar();
 function displayTable() {
     for (let transaction of transactionList) {
         displayTransaction(transaction);
     }
 }
-
+// Restricts input for the given textbox to the given inputFilter.
+function setInputFilter(textbox, inputFilter) {
+  ["input", "keydown", "keyup", "mousedown", "mouseup", "select", "contextmenu", "drop"].forEach(function(event) {
```
Comment (nickdemianchuk) [inline assets/scripts/transactions.js]: should be single quotes

File: assets/scripts/transactions.js [hunk 10]
```diff
@@ -1,13 +1,39 @@
...
+    textbox.addEventListener(event, function() {
+      if (inputFilter(this.value)) {
+        this.oldValue = this.value;
+        this.oldSelectionStart = this.selectionStart;
+        this.oldSelectionEnd = this.selectionEnd;
+      } else if (this.hasOwnProperty("oldValue")) {
```
Comment (nickdemianchuk) [inline assets/scripts/transactions.js]: single quotes here

**Review by nickdemianchuk — CHANGES_REQUESTED** [review]
please address all of the comments

File: assets/scripts/transactions.js (hunk 1 above)
Comment (asbel) [reply assets/scripts/transactions.js]: I did to one in the util folder

**Pull request closed without merging**